### PR TITLE
Add tests for BLAS.dot, BLAS.dotc, and BLAS.dotu

### PR DESCRIPTION
**Pull request by sethaxen**
As requested by @ZuseZ4, this PR just copies the tests from #739 but not the EnzymeRules.

### CODE REVIEW

**Comment by ZuseZ4** [comment]
Didn't intend to close, will merge it once I added coverage for dotu, dotc

File: test/runtests.jl [hunk 1]
```diff
@@ -76,6 +76,7 @@ include("typetree.jl")
         # XXX invalidation does not work on Julia 1.8
         include("ruleinvalidation.jl")
     end
+    include("rules/LinearAlgebra/blas.jl")
```
Comment (wsmoses) [inline test/runtests.jl:79]: Can we have this run, even if enzymerules are not supported (thus to test tablegen and fallback blas)?

Comment (wsmoses) [reply test/runtests.jl:79]: similarly can we move the test file into just tests rather than rules?

**Review by wsmoses — APPROVED** [review]

**Comment by sethaxen** [comment]
I updated this PR to use EnzymeTestUtils.

**Comment by wsmoses** [comment]
sgtm and still approved to merge (once tests pass). @ZuseZ4 was also wondering if you'd be able to add tests for some other BLAS functions as well.

**Comment by sethaxen** [comment]
Sure, which ones?

**Comment by wsmoses** [comment]
@ZuseZ4 mentioned gemv/gemm/scal/axpy/spmv

also it looks like CI fails?

**Comment by ZuseZ4** [comment]
> Sure, which ones?

I did implement scal, axpy, gemv, spmv, gemm,
since these are the relevant ones for Julia.
So tests for these would be the most helpful for me.

**Comment by sethaxen** [comment]
> also it looks like CI fails?

Pretty sure the failure is due to https://github.com/JuliaLang/Pkg.jl/issues/1585. One way to workaround this is to not have a separate `Project.toml` in `test/`. But trying a different workaround.

> I did implement scal, axpy, gemv, spmv, gemm,

Okay, I'll add this once I fix the CI

**Comment by ZuseZ4** [comment]
thanks! Will try how tablegen does against the testcases tomorrow.

**Comment by sethaxen** [comment]
> thanks! Will try how tablegen does against the testcases tomorrow.

Are the tablegen rules currently not being used? That could explain why when I added rules for `scal!`, I got a number of failures.

**Pull request merged**